engine/access/rest/middleware/request_attribute.go

```diff
@@ -0,0 +1,21 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+)
+
+type ctxKeyType string
+
+// addRequestAttribute adds the given attribute name and value to the request context
+func addRequestAttribute(req *http.Request, attributeName string, attributeValue interface{}) *http.Request {
+	contextKey := ctxKeyType(attributeName)
```

Comment:
is this necessary? Isn't this just casting a string to a string?

Reply:
oh so this is necessary to avoid this linter error:

> SA1029: should not use built-in type string as key for value; define your own type to avoid collisions (staticcheck)